File: app/src/main/java/com/anthony/deepl/openl/view/main/ShrinkSpinnerAdapter.java
package com.anthony.deepl.openl.view.main;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import com.anthony.deepl.openl.R;

public class ShrinkSpinnerAdapter<T> extends ArrayAdapter<T> {

    private Context mContext;
    private String mDetectedLanguage;

    public ShrinkSpinnerAdapter(@NonNull Context context, int resource, @NonNull T[] objects) {
        super(context, resource, objects);
        mContext = context;
    }

    @NonNull
    @Override
    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
        int selectedItemPosition = position;
        if (parent instanceof AdapterView) {
            selectedItemPosition = ((AdapterView) parent).getSelectedItemPosition();
        }
        if (selectedItemPosition == 0 && mDetectedLanguage != null) {
            return makeLayout(convertView, parent);
        }
        return super.getView(selectedItemPosition, convertView, parent);
    }

    @Override
    public View getDropDownView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
        return super.getDropDownView(position, convertView, parent);
    }

    private View makeLayout(final View convertView, final ViewGroup parent) {
        TextView tv;
        if (convertView != null) {
            tv = (TextView) convertView;
        } else {
            tv = (TextView) LayoutInflater.from(mContext).inflate(R.layout.item_language_spinner, parent, false);
        }

        tv.setText(mDetectedLanguage);
        return tv;
    }

    public void setDetectedLanguage(String detectedLanguage) {
        mDetectedLanguage = detectedLanguage;
    }

    public void clearDetectedLanguage() {
        mDetectedLanguage = null;
    }
}

File: app/src/main/java/com/anthony/deepl/openl/view/main/ShrinkSpinnerAdapter.kt
package com.anthony.deepl.openl.view.main

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.AdapterView
import android.widget.ArrayAdapter
import android.widget.TextView

import com.anthony.deepl.openl.R

class ShrinkSpinnerAdapter<T>(private val mContext: Context, resource: Int, objects: Array<T>) : ArrayAdapter<T>(mContext, resource, objects) {
    private var mDetectedLanguage: String? = null

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var selectedItemPosition = position
        if (parent is AdapterView<*>) {
            selectedItemPosition = parent.selectedItemPosition
        }
        return if (selectedItemPosition == 0 && mDetectedLanguage != null) {
            makeLayout(convertView, parent)
        } else super.getView(selectedItemPosition, convertView, parent)
    }

    private fun makeLayout(convertView: View?, parent: ViewGroup): View {
        val tv =
                if (convertView != null) {
                    convertView as TextView
                } else {
                    LayoutInflater.from(mContext).inflate(R.layout.item_language_spinner, parent, false) as TextView
                }

        tv.text = mDetectedLanguage
        return tv
    }

    fun setDetectedLanguage(detectedLanguage: String) {
        mDetectedLanguage = detectedLanguage
    }

    fun clearDetectedLanguage() {
        mDetectedLanguage = null
    }
}
